showcase/README.md

```diff
@@ -2,96 +2,149 @@
-|   |   |   |   foo.component.ts        # configuration metadata for this smart component
-|   |   |   |   foo.e2e.ts              # end-to-end test for this smart component
-|   |   |   |   index.ts                # exports the smart component
+|   |   +---foo                                      # feature module (create a folder for each feature module)
+|   |   |   |   module.ts                            # module definition (exports an NgModule)
+|   |   |   |   routes.ts                            # routing configuration of this module
+|   |   |   |   index.ts                             # barrel: re-export all module symbols
+|   |   |   |
+|   |   |   +---pages                                # smart components for this module
+|   |   |   |   |   index.ts                         # barrel: re-export all smart components
+|   |   |   |   |
+|   |   |   |   +---foo-page                         # example smart component
+|   |   |   |   |       foo-page.component.css       # styles for this smart component
+|   |   |   |   |       foo-page.component.html      # template for this smart component
+|   |   |   |   |       foo-page.component.spec.ts   # unit tests for this smart component
+|   |   |   |   |       foo-page.component.ts        # component class for this smart component
+|   |   |   |   |       index.ts                     # barrel: re-export the smart component
+|   |   |   |   |
+|   |   |   |   \---...
+|   |   |   |
+|   |   |   +---components                           # dumb components for this module
+|   |   |   |   |   index.ts                         # barrel: re-export all dumb components
+|   |   |   |   |
+|   |   |   |   +---baz                              # example dumb component
+|   |   |   |   |       baz.component.css            # styles for this dumb component
+|   |   |   |   |       baz.component.html           # template for this dumb component
+|   |   |   |   |       baz.component.spec.ts        # unit tests for this dumb component
+|   |   |   |   |       baz.component.ts             # component class for this dumb component
+|   |   |   |   |       baz.e2e.ts                   # end-to-end test for this dumb component
+|   |   |   |   |
+|   |   |   |   \---...
+|   |   |   |
+|   |   |   +---actions                              # NGRX actions utility classes for this module
+|   |   |   |       index.ts                         # barrel: re-export all actions
+|   |   |   |       foo.actions.ts                   # defines Foo action types and all Foo related action classes
+|   |   |   |       ...
+|   |   |   |
+|   |   |   +---effects                              # NGRX effects: classes that handle specific actions and react to those
+|   |   |   |       index.ts                         # barrel: re-export all effects
+|   |   |   |       foo.effects.ts                   # effects class that in reaction to actions it can interact with other services, dispatch other actions, ...
```

Comment:
maybe "effects class that can interact with other in reaction to actions"?

Reply:
Indeed, much better. I'll fix it ;)